### ISSUE TITLE
select expand(rid) from index ... bug from java api

### ISSUE THREAD

**Issue by publicocean0** [orientechnologies/orientdb#4082]
```
List<ODocument> list=db.command(new OCommandSQL("select expand(rid) from index:index_l")...
```

why return elements of class ORecordId and not ODocument (i expanded for for see also fields inside)???

**Comment by publicocean0** [comment]
if you want i see another little problem , it seams a bug:

begin
let d =  select rid from index:index_l where key = 'f1'
let c = delete from $d
commit retry 100
return $c

return 0 not 1

i tried to change row let d =  select rid from index:index_l where key = 'f1'
in let d =  select expand(rid) from index:index_l where key = 'f1'
but same result.

**Comment by luigidellaquila** [comment]
These issues were fixed with the new SQL executor (v 3.0)

Closing, thanks

Luigi
